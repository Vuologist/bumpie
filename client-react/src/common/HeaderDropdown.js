import React from "react";
import styled from "styled-components";

const Container = styled.header`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 20px;
  align-items: center;
  background-color: #8fe8df;
  color: #ff4d4d;
  padding-top: 20px;
  padding-bottom: 20px;
`;
const Card = styled.div`
  display: -webkit-box;
  display: -ms-flexbox;
  display: none;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  position: absolute;
  left: 50%;
  right: auto;
  text-align: center;
  transform: translate(-50%, 0);
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.19), 0 3px 5px 0 rgba(0, 0, 0, 0.19);
`;

const ListGroup = styled.ul`
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 0;
  margin-bottom: 0;
  border-right: 0;
  border-left: 0;
  list-style-type: none;
  border-radius: 0;
`;

const ListItem = styled.li`
  border-top: 0;
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
  border-right: 0;
  border-left: 0;
  border-radius: 0;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 12px;
  padding-bottom: 12px;
`;

const Link = styled.a`
  color: black;
`;
const HeaderDropdown = ({ className, onMouseEnter, onMouseLeave }) => (
  <Card
    className={className}
    onMouseEnter={onMouseEnter}
    onMouseLeave={onMouseLeave}
  >
    <ListGroup>
      <ListItem>
        <Link>Notification Settings</Link>
      </ListItem>
      <ListItem>
        <Link>User Settings</Link>
      </ListItem>
      <ListItem>
        <Link>Log Out</Link>
      </ListItem>
    </ListGroup>
  </Card>
);

export default HeaderDropdown;
