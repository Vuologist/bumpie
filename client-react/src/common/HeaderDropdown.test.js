import React from "react";
import HeaderDropdown from "./HeaderDropdown";
import renderer from "react-test-renderer";

it("render HeaderDropdown correctly", () => {
  const tree = renderer.create(<HeaderDropdown />).toJSON();
  expect(tree).toMatchSnapshot();
});
