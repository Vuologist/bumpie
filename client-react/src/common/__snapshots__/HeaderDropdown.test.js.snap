// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render HeaderDropdown correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0,0,0,0.125);
  position: absolute;
  left: 50%;
  right: auto;
  text-align: center;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.19),0 3px 5px 0 rgba(0,0,0,0.19);
}

.c1 {
  display: -webkit-box;
  display: -ms-flexbox;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 0;
  margin-bottom: 0;
  border-right: 0;
  border-left: 0;
  list-style-type: none;
  border-radius: 0;
}

.c2 {
  border-top: 0;
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
  border-right: 0;
  border-left: 0;
  border-radius: 0;
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 12px;
  padding-bottom: 12px;
}

.c3 {
  color: black !important;
}

@media (max-width:425px) {
  .c0 {
    left: 0%;
  }
}

<div
  className="c0"
>
  <ul
    className="c1"
  >
    <li
      className="c2"
    >
      <a
        className="c3"
      >
        Notification Settings
      </a>
    </li>
    <li
      className="c2"
    >
      <a
        className="c3"
      >
        User Settings
      </a>
    </li>
    <li
      className="c2"
    >
      <a
        className="c3"
      >
        Log Out
      </a>
    </li>
  </ul>
</div>
`;
